### ISSUE TITLE
Internal exception when loading many CSV files

### ISSUE THREAD

**Issue by ivan-tkatchev** [duckdb/duckdb#11843]
### What happens?

Loading many compressed CSV files via glob mask causes an internal exception in duckdb:

```
>>> duckdb.query("copy (select *, cast(to_timestamp(time) as date) as date from read_csv('s3://api-data/metrics/*', ignore_errors=true)) to 's3://api-data/metrics/merged' (format parquet, partition_by date, filename_pattern '0_{i}')")
  0% ▕                                                            ▏ Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
duckdb.duckdb.InternalException: INTERNAL Error: CSV Buffer is not available to reconstruct CSV Line, please open an issue with your query and dataset.
```

duckdb built from source, revision 86fee9ed386eaa2d50a7154416ebb824980e8d59

Loading one individual file works correctly:

```
>>> duckdb.query("copy (select *, cast(to_timestamp(time) as date) as date from read_csv('s3://api-data/metrics/ts-1714119379-0000336915-metrics.gz')) to 's3://api-data/metrics/merged' (format parquet, partition_by date, filename_pattern '1_{i}', overwrite_or_ignore true)")
100% ▕████████████████████████████████████████████████████████████▏ 
```

### To Reproduce

 ¯\_(ツ)_/¯

### OS:

Linux

### DuckDB Version:

86fee9ed386eaa2d50a7154416ebb824980e8d59

### DuckDB Client:

Python

### Full Name:

Ivan Tkachev

### Affiliation:

Hyper AdTech

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a nightly build

### Did you include all relevant data sets for reproducing the issue?

No - I cannot easily share my data sets due to their large size

### Did you include all code required to reproduce the issue?

- [ ] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [ ] Yes, I have

**Comment by Tishj** [comment]
If there is no privacy concern we'd very much like it if you could attach the dataset, uploading through WeTransfer or SendBig if WeTransfer is blocked in your region

**Comment by ivan-tkatchev** [comment]
Sure thing. I shared a SendBig link to the email I found in your github profile.
(Approximately 7 Gb of compressed CSV.)

**Comment by pdet** [comment]
Hi @ivan-tkatchev 
Thanks for opening the issue and sharing the data.

Unfortunately, I was not able to reproduce this locally on the latest `main`. Could you try that and let me know if you still experience this issue?

**Comment by brianwyka** [comment]
I've encountered a similar issue on `0.10.2`.  Example query:

```sql
SELECT DISTINCT concat(column02, '@D@', column14)
FROM read_csv('s3://bucket/path/**/*', ignore_errors=true)
WHERE column14 <> '' AND column14 IS NOT NULL
```

The files in the bucket are `gzip` compressed `TSV` files.  There are a total of about `20` files in the s3 path being scanned.

I tried removing the `DISTINCT` and `concat` to see if it makes a difference and no luck...

**Comment by miladhatami1393** [comment]
Thanks for opening the issue and sharing the data.

**Comment by pdet** [comment]
I believe this issue has been fixed in the `main` branch, next week's  `v1.1`.

If anyone still encounters this, please open a new issue with the data and a reproducible example (e.g., queries).

**Comment by sarda-devesh** [comment]
I ran into the same issue on a private dataset - Fixed the issue by upgrading to `duckdb-1.1.0`
